### ISSUE TITLE
Fix Synchronous vs Async Hooks Documentation

### ISSUE THREAD

**Issue by nickw-adonis** [payloadcms/payload#7930]
The current documentation on async vs synchronous hooks swapped the description of behaviors. Specifically, the async definition

> Async hooks are run in series - so if you have two async hooks defined, the second hook will wait for the first to complete before it starts

is probably swapped with the synchronous definition
> If your Hook simply performs a side-effect, such as updating a CRM, it might be okay to define it synchronously, so the Payload operation does not have to wait for your hook to complete.

[Here is the documentation i'm referencing](https://payloadcms.com/docs/hooks/overview#async-vs-synchronous)

### Suggested Correction:
Async Hooks:
- Usage: When a hook performs operations involving asynchronous actions, such as fetching data from an external source.
- Behavior: Async hooks can run concurrently, meaning multiple async operations can be initiated simultaneously without waiting for each to complete sequentially.
- Benefit: Improves performance by not blocking the main operation, allowing multiple tasks to be handled in parallel.

Synchronous Hooks:
- Usage: Suitable for hooks that need to complete their operations before the main operation can proceed, ensuring that tasks are performed in a specific order.
- Behavior: These hooks run in a blocking manner, causing tasks to be executed one after the other. The main operation waits for each synchronous hook to finish before moving on.
- Benefit: Ensures that all necessary operations are completed in a defined sequence, maintaining order and consistency.

### Revised Documentation Example:

> Async vs. synchronous
> 
> All hooks can be written as either synchronous or asynchronous functions. If the hook relies on asynchronous actions such as fetching data from a third party, it might make sense to define your hook as an asynchronous function. This allows the main operation to continue without waiting for your hook to complete, making the process more efficient. Async hooks can run concurrently, meaning multiple async hooks can execute simultaneously without waiting for one another.
> 
> If your hook needs to modify data before a document is updated or created and should complete before the operation's lifecycle continues, it might be better to define it as a synchronous function. Synchronous hooks run in a blocking manner, meaning that the main operation will wait for each hook to complete before moving on to the next step.
>

**Comment by paulpopus** [comment]
Ok so in the paradigm of javascript you would be right about explaining the difference between the synchronous and asynchronous hooks.

However in Payload the difference is more related to how we internally await and run these hooks. Asynchronous hooks don't execute simultaneously as that could cause quite a myriad of problems when it comes to data manipulation.

See this file for example, how beforeChange hooks are run in an update operation https://github.com/payloadcms/payload/blob/beta/packages/payload/src/collections/operations/update.ts#L260

This reducer goes through every hook and awaits the result before executing the next one.

So in the context of Payload, most of your hooks will be asynchronous. However if you don't want Payload to wait for your function to finish then you can make it synchronous and this is more like a "fire and forget", useful for things like sending off a notification, things not needed for the rest of the hooks to proceed.

Hope this makes sense, so yeah this bit in the docs is Payload specific.

**Comment by nickw-adonis** [comment]
Got it, thanks!
